Shopify/cli
# Enable draft pushing for UI extensions when using unified deploys

### WHY are these changes introduced?

Fixes https://github.com/Shopify/app-deploys/issues/612



Currently, if someone `dev`s an extension that hasn't ever been deployed, it won't show up at all in the browser, unless they've opened via the dev console. This is a confusing experience.

We'd prefer that a draft of the extension is always pushed (including watching for changes), so that the developer sees a reasonably consistent experience just from opening the store directly vs. opening via the dev preview URL.

### WHAT is this pull request doing?



Pushes UI extension drafts as files are updated, so dev console and standard views of the extension will look the same, creating a more intuitive and consistent experience across extension types.

The only difference is that there's no HMR unless someone opens the dev console. I think that's acceptable, and much better than only updating for the local development environment. And if they refresh the page, the new version will be available.

### How to test your changes?



In all steps, be sure **never** to open the dev panel. Otherwise, the experiment is tainted and you need to start over.

Bold is changed behavior.

1. On `main`, create an app with unified deployments active.
2. Run `pnpm shopify app dev --path /path/to/your/app`, open the preview URL, and install the app on your store.
3. Generate a product subscription UI extension but DO NOT DEPLOY.
4. Run `pnpm shopify app dev --path /path/to/your/app`.
5. Log into Partners and find your app. On the extensions page, you should see that drafts haven't been pushed.
6. Quit `dev` and check out this branch.
7. Run `pnpm shopify app dev --path /path/to/your/app`.
8. **On the extensions page, you should now see that a draft has been pushed. Additionally, when updating a file, you should see an update to the last time a draft was pushed.**
9. Kill `dev` and run `pnpm shopify app deploy --path /path/to/your/app`.
10. Open your store in the browser, and navigate to a product. You should now see the subscription extension.
11. Run `pnpm shopify app dev --path /path/to/your/app`.
12. Make a change to your extension locally and save.
13. Refresh the page in your store, and you should see the updates applied.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
